src/sentry/static/sentry/app/views/settings/components/forms/selectField.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import {OptionsType, ValueType} from 'react-select';
+
+import SelectControl, {ControlProps} from 'app/components/forms/selectControl';
+import {Choices, SelectValue} from 'app/types';
+import InputField from 'app/views/settings/components/forms/inputField';
+
+type InputFieldProps = React.ComponentProps<typeof InputField>;
+
+type Props<OptionType> = InputFieldProps &
+  Omit<ControlProps<OptionType>, 'onChange'> & {
+    /**
+     * Should the select be clearable?
+     */
+    allowClear?: boolean;
+    /**
+     * Should the select allow empty values.
+     */
+    allowEmpty?: boolean;
+    small?: boolean;
+    /**
+     * A label that is shown inside the select control.
+     */
+    inFieldLabel?: string;
+    stacked?: boolean;
+  };
+
+function getChoices<T>(props: Props<T>): Choices {
+  const choices = props.choices;
+  if (typeof choices === 'function') {
+    return choices(props);
```

Comment:
should we omit `choices` in this `props`?

Reply:
I didn't want to change the runtime code for this. I'd like to remove `choices` entirely but there is more work to do there.c